### PR TITLE
DAY-2: Add full skeleton for product service and product management

### PR DESCRIPTION
**Pull request by BrianZunigaEncora**
Added product management module with the rest services with no connection to queue or data base.
code coverage 100%

### CODE REVIEW

File: productService/src/main/java/com/skytouch/productservice/service/ProductServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.skytouch.productservice.service;
+
+import com.skytouch.dto.ProductDTO;
+
```
Comment (juan-caravantes) [inline productService/src/main/java/com/skytouch/productservice/service/ProductServiceImpl.java:6]: `@Service`

File: productService/src/main/java/com/skytouch/productservice/service/ProductServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+public class ProductServiceImpl implements ProductService {
+    @Override
```
Comment (juan-caravantes) [inline productService/src/main/java/com/skytouch/productservice/service/ProductServiceImpl.java:11]: `@Transactional`

File: productsManagement/src/main/java/com/productsmanagement/controller/ProductManagementController.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package com.productsmanagement.controller;
+
+import com.productsmanagement.service.ProductManagementService;
+import com.skytouch.dto.ProductDTO;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
```
Comment (juan-caravantes) [inline productsManagement/src/main/java/com/productsmanagement/controller/ProductManagementController.java]: This controller will be consumed by JSPs, are you sure that `@RestController` is the correct annotation you need here?

File: productsManagement/src/main/java/com/productsmanagement/controller/ProductManagementController.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        this.productManagementService = productManagementService;
+    }
+
+    @GetMapping("/products")
+    public List<ProductDTO> getProducts() {
+        return productManagementService.getAllProducts();
+    }
+
+    @PostMapping(value = "/products", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ProductDTO addProduct(@RequestBody ProductDTO productDTO) {
```
Comment (juan-caravantes) [inline productsManagement/src/main/java/com/productsmanagement/controller/ProductManagementController.java]: `@RequestBody` is not needed when using `@RestController`.

File: commons/src/main/java/com/skytouch/model/ProductModel.java
Comment (juan-caravantes) [inline commons/src/main/java/com/skytouch/model/ProductModel.java:1]: What is the difference between ProductDTO and ProductModel?

File: productsManagement/src/main/java/com/productsmanagement/controller/ProductManagementController.java (hunk 3 above)
Comment (BrianZunigaEncora) [reply productsManagement/src/main/java/com/productsmanagement/controller/ProductManagementController.java]: I wanted to see if that was working using my postman, Im planning to change this to controller in order to use these services along JSP

**Pull request merged**